### DISCUSSION TITLE
Add support for safetensor format

### DISCUSSION THREAD

**Discussion by hassan-sd** [civitai/civitai#59] [Ideas]
We are limited to upload cktp files only but we have the ability to save in other formats, more recently the SafeTensor format.

Allow the uploading of other formats so users can opt in to choose the version and format they prefer.

In this case, having multiple model uploads for each model version is desirable.

**Comment by JustMaier** [comment]
Since `ckpt`s can easily be converted to `safetensor`s I wonder if it makes sense for everything to just be available as both until the standard changes to `safetensor`. Currently we scan all of the models that people upload so maybe it makes sense to convert and reupload as whatever format is missing upon scan.

I think the only challenge might be the memory requirements. Our scanner currently only runs on 1GB RAM.

**Comment by JustMaier** [comment]
**Support for `safetensor` is live!**
- You can select a `safetensor` file when uploading model versions
- Downloads now display the type of file they are. As part of this we've also redesigned how we show if something is verified:
![Demo](https://user-images.githubusercontent.com/607609/205387831-8bff07b2-6ed7-4c67-817c-9e07e47f6446.png)
